packages/block-editor/src/components/keyboard-shortcuts/index.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * External dependencies
+ */
+import { first, last } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect, useCallback } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useShortcut } from '@wordpress/keyboard-shortcuts';
+import { __ } from '@wordpress/i18n';
+
+function KeyboardShortcuts() {
+	const target = useRef( null );
+
+	// Registering the shortcuts
+	const { registerShortcut } = useDispatch( 'core/keyboard-shortcuts' );
+	useEffect( () => {
+		registerShortcut(
+			'core/block-editor/duplicate',
+			'block',
+			__( 'Duplicate the selected block(s).' ),
+			{
+				modifier: 'primaryShift',
+				character: 'd',
+			}
+		);
+
+		registerShortcut(
+			'core/block-editor/remove',
+			'block',
+			__( 'Remove the selected block(s).' ),
+			{
+				modifier: 'access',
+				character: 'z',
+			}
+		);
+
+		registerShortcut(
+			'core/block-editor/insertBefore',
```

Comment:
Just wondering about the mixture of camel-case and kebab-case. Is the idea that `insertBefore` matches the function name of an action as much as possible?

Reply:
Good call, I'll rename to camel-case.